Content.Server/AlertLevel/ChangeAlertLevelSystem.cs

```diff
@@ -0,0 +1,63 @@
+using Content.Shared.Interaction;
+using Content.Shared.ChangeAlertLevel;
+using Content.Server.AlertLevel;
+using Content.Server.Audio;
+using Content.Server.Chat.Systems;
+using Content.Server.Explosion.EntitySystems;
+using Content.Server.Pinpointer;
+using Content.Server.Popups;
+using Content.Server.Station.Systems;
+using Content.Shared.Audio;
+using Content.Shared.Containers.ItemSlots;
+using Content.Shared.Coordinates.Helpers;
+using Content.Shared.DoAfter;
+using Content.Shared.Examine;
+using Content.Shared.Maps;
+using Content.Shared.Nuke;
+using Content.Shared.Popups;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Containers;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+
+namespace Content.Server.ChangeAlertLevel;
+
+public sealed class ChangeAlertLevelSystem : EntitySystem
+{
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly AlertLevelSystem _alertLevel = default!;
+    [Dependency] private readonly StationSystem _station = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SharedChangeAlertLevelComponent, ActivateInWorldEvent>(OnActivated);
+    }
+
+    public void OnActivated(Entity<SharedChangeAlertLevelComponent> ent, ref ActivateInWorldEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        var buttonXform = Transform(ent.Owner);
+        var stationUid = _station.GetStationInMap(buttonXform.MapID);
+
+        var playsound = true;
+        var announce = true;
+        var force = false;
+        var locked = false;
+
+        if (stationUid != null)
+            _alertLevel.SetLevel(stationUid.Value, ent.Comp.AlertLevelOnActivate, playsound, announce, force, locked);
+        Dirty(ent.Owner, ent.Comp);
+
+        _audio.PlayPvs(ent.Comp.ClickSound, ent.Owner, AudioParams.Default.WithVariation(0.125f).WithVolume(8f));
+
+        args.Handled = true;
```

Comment:
this is the reason you cant use signal button, either that system or this one will handle the interaction event and prevent the other system from working.

make an event say `ButtonPressedEvent` and raise it in the signal button's system and then in here have all this file just handle it like this

```cs
private void OnButtonPressed(...)
{
     if (_station.GetStationInMap(Transform(ent).MapID) is {} station)
         _alertLevel.SetLevel(station, ent.Comp.AlertLevelOnActivate, playsound: true, announce: true, force: false, locked: false);
}
```

also this should probably use GetOwningStation to support multiple stations in the same map, ask sloth

with these changes the component and system can be renamed to `AlertLevelOnPress` which makes it easier to immediately understand what it does compared to `ChangeAlertLevel`

Reply:
alright, will probably get to this tomorrow, thanks for all the help!